actions/src/sema4ai/actions/chat/_client.py

```diff
@@ -156,65 +153,69 @@ def set_bytes(
             }
-            data = json.dumps(
-                {"file_name": filename, "file_size": len(content)}
-            ).encode("utf-8")
+            payload = {
+                "file_name": filename,
+                "file_size": len(content)
+            }
 
-            # Send the initial request
-            response = sema4ai_http.post(url, body=data, headers=headers)
+            # Request upload URL from server
+            response = sema4ai_http.post(
+                url,
+                json=payload,
+                headers=headers
+            )
             self._raise_for_status(
                 f"Failed when requesting upload for file {filename} to {self._url}.",
                 response,
                 (200,),
             )
 
             response_data = response.json()
-            file_url = response_data["url"]
+            upload_url = response_data["url"]
             file_id = response_data["file_id"]
             file_ref = response_data["file_ref"]
 
-            parsed_url = urllib.parse.urlparse(file_url)
-            if parsed_url.scheme == "file":
-                # We need to add the file to the local file system
-                p = Path(_uris.to_fs_path(file_url))
-                p.parent.mkdir(parents=True, exist_ok=True)
-                p.write_bytes(content)
+            # If the URL is a local path, write directly to filesystem
+            if upload_url.startswith('/'):
+                # It's a local filesystem path
+                import os
+                os.makedirs(os.path.dirname(upload_url), exist_ok=True)
+                with open(upload_url, 'wb') as f:
+                    f.write(content)
             else:
-                # We have a presigned-post URL, upload to it!
-                response = sema4ai_http.post(
-                    file_url,
-                    fields={**response_data["form_data"], "file": (file_ref, content)},
-                    headers={"Content-Type": content_type},
+                # Remote URL case
+                upload_response = sema4ai_http.post(
+                    upload_url,
+                    fields=response_data["form_data"],
+                    files={"file": (filename, content, content_type)},
                 )
                 self._raise_for_status(
-                    f"Failed when uploading file {filename} to {file_url}.",
-                    response,
+                    f"Failed when uploading file {filename} to {upload_url}.",
+                    upload_response,
                     (200, 204),
                 )
 
-            # Send the upload complete notification
-            url = f"{self._url}/threads/{thread_id}/files/confirm-upload"
-            headers = {
-                "Content-Type": "application/json",
-                "x-action-context": self.get_x_action_context(),
+            # Confirm upload
+            confirm_url = f"{self._url}/api/v1/threads/{thread_id}/files/confirm-upload"
```

Comment:
The `/api/v1` needs to be passed from the agent server (so, the old thing was correct).